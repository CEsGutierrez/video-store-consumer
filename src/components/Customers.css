.customersList {
  display: flex;
  flex-wrap: wrap;
}