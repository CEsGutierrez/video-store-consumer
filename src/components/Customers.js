import React, {useEffect} from 'react';
import Customer from './Customer';
import PropTypes from 'prop-types';
import './Customers.css';


const parseCustomers = (customers, selectCustomerCallback) => {
  return customers.map((customer) => {
    return <Customer
      key={customer.id}
      {...customer}
      selectCustomerCallback={selectCustomerCallback}
      />
  });
}

const Customers = ({customerList, selectCustomerCallback, resetMessageCallback}) => { // selectCustomerCallback
  useEffect(() => {
    resetMessageCallback()
  }, [resetMessageCallback]);

  return(
    <div>
      <h2 className='pageHeader'> Customers </h2>
      <div className='customersList'>
        {parseCustomers(customerList, selectCustomerCallback)}
      </div>
    </div>
  )
}

Customers.propTypes = {
  customerList: PropTypes.array.isRequired,
  selectCustomerCallback: PropTypes.func.isRequired,
  resetMessageCallback: PropTypes.func.isRequired,
};

export default Customers;