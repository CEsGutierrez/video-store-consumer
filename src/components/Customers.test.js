import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Customers from './Customers';

describe('Customers', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Customers
      customerList={[]}
      selectCustomerCallback={() => { }}
      resetMessageCallback={() => { }}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });

});