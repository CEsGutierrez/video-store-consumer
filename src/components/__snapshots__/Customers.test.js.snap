// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customers that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="pageHeader"
    >
       Customers 
    </h2>
    <div
      class="customersList"
    />
  </div>
</DocumentFragment>
`;
